cargo-marker/src/observability/display.rs

```diff
@@ -0,0 +1,47 @@
+mod cli;
+
+pub(crate) use cli::*;
+
+use itertools::Itertools;
+use std::fmt;
+use yansi::Paint;
+
+/// Lightweight and dumb highlighter for TOML. It doesn't actually parse TOML
+/// correctly, but it's good enough for simple messages that need to present
+/// an example TOML snippet.
+pub(crate) fn toml(toml: &str) -> String {
+    toml.lines()
+        .map(|line| {
+            let punct = ['[', ']', '{', '}', '='];
+
+            let line = punct.into_iter().fold(line.to_string(), |line, punct| {
+                line.replace(punct, &punct.yellow().to_string())
+            });
+
+            if line.trim().is_empty() {
+                return line.to_string();
+            }
+
+            if line.starts_with('#') {
+                return line.black().to_string();
+            }
+
+            if let Some((key, val)) = line.split_once('=') {
+                return format!("{}{}{}", key.cyan().bold(), "=".yellow(), val.blue());
+            }
+
+            line.to_string()
+        })
+        .join("\n")
+}
+
+/// Displays a stage of work that `cargo-marker` performs.
+/// This is specifically formatted to be aligned with the other cargo
+/// output like `Compiling` or `Checking`.
+pub(crate) fn print_stage(name: &str) {
+    println!("      {} {}", "Marker".bold().green(), stage(name));
+}
```

Comment:
Optional suggestion, an empty line, before the stage, would make the output more readable for me:

```
      Marker compiling rustc driver
   Compiling rustc_tools_util v0.3.0
   Compiling libloading v0.8.0
   Compiling bumpalo v3.13.0
   Compiling marker_api v0.3.0-dev ([...]/marker/marker_api)
   Compiling marker_rustc_driver v0.3.0-dev ([...]/marker/marker_rustc_driver)
   Compiling marker_utils v0.3.0-dev ([...]/marker/marker_utils)
   Compiling marker_adapter v0.3.0-dev ([...]/marker/marker_adapter)
    Finished dev [unoptimized + debuginfo] target(s) in 7.72s

      Marker compiling lints
    Updating crates.io index
   Compiling proc-macro2 v1.0.66
   Compiling unicode-ident v1.0.11
   Compiling syn v1.0.109
   Compiling quote v1.0.33
   Compiling visibility v0.0.1
   Compiling marker_api v0.3.0-dev ([...]/marker/marker_api)
   Compiling marker_lints v0.3.0-dev ([...]/marker/marker_lints)
    Finished release [optimized] target(s) in 4.33s

      Marker linting
    Checking marker_api v0.3.0-dev ([...]/marker/marker_api)
   Compiling marker_rustc_driver v0.3.0-dev ([...]/marker/marker_rustc_driver)
    Checking marker_error v0.3.0-dev ([...]/marker/marker_error)
    Checking marker_uitest v0.3.0-dev ([...]/marker/marker_uitest)
```

```suggestion
pub(crate) fn print_stage(name: &str) {
    println!("\n      {} {}", "Marker".bold().green(), stage(name));
}
```